src/connector/src/sink/batching_log_sink.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use async_trait::async_trait;
+
+use crate::sink::file_sink::opendal_sink::OpenDalSinkWriter;
+use crate::sink::log_store::{LogStoreReadItem, TruncateOffset};
+use crate::sink::writer::SinkWriter;
+use crate::sink::{LogSinker, Result, SinkLogReader};
+
+/// `BatchingLogSinkerOf` is used for a commit-decoupled sink that supports cross-barrier batching.
+/// Currently, it is only used for file sinks, so it contains an `OpenDalSinkWriter`.
+/// When the file sink writer completes writing a file, it truncates the corresponding `chunk_id` in the log store.
+pub struct BatchingLogSinkerOf {
```

Comment:
nits: rename to `BatchingLogSinkerOf` because it contains a concrent writer type

Reply:
typo?